SmartDeviceLink-iOS/SmartDeviceLink/SDLProtocolHeader.m

```diff
@@ -38,20 +38,27 @@ - (NSString *)description {
     return description;
 }
 
-
 + (SDLProtocolHeader *)headerForVersion:(UInt8)version {
-    if (version == 1) {
-        return [[SDLV1ProtocolHeader alloc] init];
-    }
-
-    if (version == 2) {
-        return [[SDLV2ProtocolHeader alloc] init];
-    }
-
     // TODO: some error handling here if unknown version is asked for,
     // but that needs to be balanced against future proofing. i.e. V3.
     // Requirements around V3 are as yet undefined so give a V2 header
-    return [[SDLV2ProtocolHeader alloc] init];
+    // lol wat.
+    switch (version) {
+        case 0: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:2];
+        } break;
+        case 1: {
+            return [[SDLV1ProtocolHeader alloc] init];
+        } break;
+        case 2:
+        case 3:
+        case 4: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:version];
+        } break;
+        default: {
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:2];
+        } break;
+    }
 }
```

Comment:
Why do you have to init a version specific object WITH a version number? These methods probably shouldn't even exist. And it might be better to have any invalid version numbers (case 0 and default) not just continue happily.